check/src/typecheck.rs

```diff
@@ -727,9 +741,9 @@ impl<'a> Typecheck<'a> {
                 let mut match_type = self.remove_alias(match_type);
                 let mut types = Vec::new();
                 let new_type = match *match_type {
```

Comment:
Pretty sure this match block can be simplified now that row polymorphism is implemented. The reason for the split is just because omitting fields from patterns was allowed previously (and still will be) but since `let { x } = .. ` would look like ``{ x : a }` it could not simply be unified with, say `{ x : Int, y : Int }` but now it can simply be unified with a polymorphic record.